.github/workflows/ci.yml

```diff
@@ -0,0 +1,166 @@
+name: CI
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+on:
+  pull_request:
+  merge_group:
+
+jobs:
+  backward_compat:
+    name: "Backward Compatibility"
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ./pytest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v4
+        with:
+          python-version: 3.11
+          cache: pip
+      - uses: Swatinem/rust-cache@a95ba195448af2da9b00fb742d14ffaaf3c21f43
+        with:
+          prefix-key: "0" # change this to invalidate CI cache
+          save-if: "false" # use the cache from nextest, but don’t double-save
+      - run: pip3 install --user -r requirements.txt
+      - run: python3 tests/sanity/backward_compatible.py
+
+  cargo_nextest:
+    name: "Cargo Nextest"
```

Comment:
Ah, just noticed, but I think you will need to add `${{ matrix.name }}` to this name, otherwise all of the different matrix elements will read "Cargo Nextest" and it isn't clear to me if all of them will need to pass, or will only one of them passing suffice to make this pass?